addons/controllers/addon_controller.go

```diff
@@ -115,10 +116,11 @@ func (r *AddonReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ct
 		return ctrl.Result{}, err
 	}
 
-	tkrName := util.GetTKRNameForCluster(ctx, r.Client, cluster)
-	if tkrName == "" {
-		log.Info("cluster does not have an associated TKR")
-		return ctrl.Result{}, nil
+	tkrName := cluster.Labels[constants.TKRLabel]
+	tkr, err := util.GetTKRByName(ctx, r.Client, tkrName)
```

Comment:
We need to add a check to `GetTKRByName()` that the returned error is not `IsNotFound(err)`:
```go
	if err := c.Get(ctx, tkrNamespaceName, tkr); err != nil && !apierrors.IsNotFound(err) {
		return nil, err
	}
```

Reply:
I just changed with a minor difference that I return nil in ```GetTKRByName()``` for the object on ```apierrors.IsNotFound(err)``` to easier check this case in the reconcile functions